src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBranchFilter.java

```diff
@@ -26,17 +26,19 @@
 
 import hudson.Extension;
 import hudson.model.Descriptor;
+import hudson.model.TopLevelItem;
+import hudson.model.View;
 import hudson.views.ViewJobFilter;
-import jenkins.branch.Branch;
-import jenkins.branch.SimpleViewBranchFilter;
+import java.util.List;
+import jenkins.scm.api.SCMHead;
 import org.kohsuke.stapler.DataBoundConstructor;
 
 /**
- * A {@link ViewJobFilter} that matches {@link PullRequestSCMHead} based branches.
+ * A {@link ViewJobFilter} that matches {@link BranchSCMHead} based branches.
  *
  * @since FIXME
  */
-public class GitHubBranchFilter extends SimpleViewBranchFilter {
+public class GitHubBranchFilter extends ViewJobFilter {
```

Comment:
If you created a view at the root of Jenkins that included recursively all children... you might have been using the GHOF version of this class